monero/wallet_client.go

```diff
@@ -415,32 +431,22 @@ func validateMonerodConfig(env common.Environment, monerodHost string, monerodPo
 // createWalletRPCService starts a monero-wallet-rpc instance. Default values are assigned
 // to the MonerodHost, MonerodPort, WalletPort and LogPath fields of the config if they
 // are not already set.
-func createWalletRPCService(conf *WalletClientConf) (*os.Process, error) {
+func createWalletRPCService(conf *WalletClientConf) (*os.Process, *common.MoneroNode, error) {
 	walletRPCBin := conf.MoneroWalletRPCPath
 	if walletRPCBin == "" {
 		var err error
 		walletRPCBin, err = getMoneroWalletRPCBin()
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 	}
 
-	if conf.MonerodHost == "" {
-		conf.MonerodHost = "127.0.0.1"
-	}
-	if conf.MonerodPort == 0 {
-		switch conf.Env {
-		case common.Mainnet, common.Development:
-			conf.MonerodPort = common.DefaultMoneroDaemonMainnetPort
-		case common.Stagenet:
-			conf.MonerodPort = common.DefaultMoneroDaemonStagenetPort
-		default:
-			panic("unhandled environment value")
-		}
+	if len(conf.MonerodNodes) == 0 {
+		conf.MonerodNodes = common.ConfigDefaultsForEnv(conf.Env).MoneroNodes
```

Comment:
`swapd`, at least right now, is always setting `conf.MonerodNodes`, but tests that don't want to configure all the values use this.